### PR TITLE
Stevenic/storage component

### PR DESCRIPTION
**Pull request by Stevenic**
Added new Strorage components and sample app.

### CODE REVIEW

File: libraries/Bot.Builder.Community.Components.Storage/readme.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+## CallDialogs Component for Bot Framework Composer
```
Comment (garypretty) [inline libraries/Bot.Builder.Community.Components.Storage/readme.md:1]: Typo, should be storage.

File: samples/components/CallDialogsSample/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# prevent appsettings.json get checked in
+**/appsettings.json
```
Comment (garypretty) [inline samples/components/CallDialogsSample/.gitignore:2]: I thought this was already excluded by the repo gitignore - not a problem right now. I have an action to consolidate both gitignore and also StyleCop rules across the Community at some point, so we can leave this for now.

**Review by garypretty — CHANGES_REQUESTED** [review]
Typo in Storage readme title.

@Stevenic do we expect we might add additional items to this package in the future? I am wondering if we should make the name more specific, like "ReadWrite Storage Component" or something, given we will hopefully see a lot of packages created in the future. Not saying we should change this, just a note for discussion.

**Comment by Stevenic** [comment]
> @Stevenic do we expect we might add additional items to this package in the future? I am wondering if we should make the name more specific, like "ReadWrite Storage Component" or something, given we will hopefully see a lot of packages created in the future. Not saying we should change this, just a note for discussion.

It's possible that more components will get added. For example the underlying storage provider allows for batch reads & writes which I'm not currently exposing.

I called this component Storage because the actions are tied directly to working with the bots IStorage provider.

File: samples/components/CallDialogsSample/.gitignore (hunk 2 above)
Comment (Stevenic) [reply samples/components/CallDialogsSample/.gitignore:2]: This gets added by composer. I didn't modify anything that composer generates.

**Comment by Stevenic** [comment]
I removed all of the copyright headers from this branch. Just apply it after the other PR and it should essentially update that sample as well

**Pull request merged**